### PR TITLE
Bindings fixes and improvements

### PR DESCRIPTION
**Pull request by flysand7**
This PR implements a few fixes and improvements for `odin-freetype` package. In particular:

- Fix the indentation on the first indented line (it was spaces while the rest of the file was tabs which confused my editor and I didn't notice until I looked at the diff of the first commit)
- Adds linking to linux (to the version installed in the system)
- Changes the `Error` type to an enumeration, added error codes.
- Changes pointers to multipointers in a few places, making them accessible like arrays.
- Fixed the TODO on glyph formats by adding the definitions of the formats.
- Added functions: `FT_GetCharIndex` and `FT_LoadGlyph`.
- Fixed the layout of the `Face_Rec` struct. `available_sizes` should have been a pointer.

Tested a simple usecase on my machine:

image

### CODE REVIEW

File: freetype.odin [hunk 1]
```diff
@@ -144,10 +246,10 @@ Face_Rec :: struct {
     style_name          : cstring,
 
     num_fixed_sizes     : c.int,
-    available_sizes     : Bitmap_Size,
+    available_sizes     : [^]Bitmap_Size,
 
     num_charmaps        : c.int,
-    charmaps            : ^Char_Map,
+    charmaps            : [^]^Char_Map_Rec,
```
Comment (englerj) [inline freetype.odin]: The addition of the multipointer here makes sense, however, I believe the type should still be `Char_Map`. `Char_Map` is itself just an alias of `^Char_Map_Rec` and it lines up with the type used in FreeType's source code.

https://gitlab.freedesktop.org/freetype/freetype/-/blob/VER-2-11-0/include/freetype/freetype.h#L1074

```suggestion
    charmaps            : [^]Char_Map,
```

**Comment by englerj** [comment]
So, this is awkward, but the spacing is actually mostly using spaces, not tabs (I think the foreign imports were using tabs and everything else were using spaces). Ideally, it should _probably_ all be using tabs since that seems to be the convention with Odin (Please correct me if I'm mistaken). 

That said, I'd really like to avoid mixing the two if possible. If you'd like to change it all to using tabs, feel free, but please don't feel like you have to. Otherwise, please remove the tabs in favor of spaces for now.

**Comment by flysand7** [comment]
You're correct, using tabs is the convention for the Odin compiler and the core libraries, I don't mean to impose either way. Looking over the diff again seems like I've forgotten to push the commit that changes the indentation to one way (if I remember correctly, it was spaces since that's what has been used, but I don't remember anymore).

Edit: pushed the missing commit

**Comment by englerj** [comment]
Looks good. Thank you for the PR.

**Pull request merged**